zerver/tests/test_auth_backends.py

```diff
@@ -3301,12 +3301,69 @@ def test_login_success(self) -> None:
             self.assertEqual(result.status_code, 302)
             self.assert_logged_in_user_id(user_profile.id)
 
+    @override_settings(REMOTE_USER_AUTH_USER_HEADER="X-Remote-User")
+    def test_http_header_login_success(self) -> None:
+        user_profile = self.example_user('hamlet')
+        email = user_profile.delivery_email
+        with self.settings(AUTHENTICATION_BACKENDS=('zproject.backends.ZulipRemoteUserBackend',)):
+            result = self.client_post('/accounts/login/sso/', HTTP_X_REMOTE_USER=email)
+            self.assertEqual(result.status_code, 302)
+            self.assert_logged_in_user_id(user_profile.id)
+
+    @override_settings(REMOTE_USER_AUTH_USER_HEADER="X-Remote-User")
+    @override_settings(REMOTE_USER_AUTH_SECRET_HEADER="X-Remote-Secret")
+    @override_settings(REMOTE_USER_AUTH_SECRET="HTTP-SECRET")
+    def test_http_header_login_success_with_secret(self) -> None:
+        user_profile = self.example_user('hamlet')
+        email = user_profile.delivery_email
+        with self.settings(AUTHENTICATION_BACKENDS=('zproject.backends.ZulipRemoteUserBackend',)):
+            result = self.client_post('/accounts/login/sso/', HTTP_X_REMOTE_USER=email, HTTP_X_REMOTE_SECRET="HTTP-SECRET")
```

Comment:
I think all the `client_post`s in these tests should be `client_get`s to be consistent with the rest of the tests in this class, and because GET is  a more natural way to access this endpoint than POST?